src/main/js/store/actions/distributionConfigs.js

```diff
@@ -241,3 +281,24 @@ export function checkDescriptorForGlobalConfig(errorFieldName, descriptorName) {
         }).catch(console.error);
     };
 }
+
+export function validateDistributionJob(config) {
+    return (dispatch, getState) => {
+        dispatch(validatingJob());
+        const { csrfToken } = getState().session;
+        const request = ConfigRequestBuilder.createValidateRequest(ConfigRequestBuilder.JOB_API_URL, csrfToken, config);
+        request.then((response) => {
+            response.json()
+                .then((responseData) => {
+                    const handler = createErrorHandler(DISTRIBUTION_JOB_VALIDATE_ERROR, () => validateJobError(responseData));
+                    if (responseData.errors && !Object.keys(responseData.errors).length) {
+                        dispatch(validatedJob());
+                    } else if (!response.ok) {
+                        dispatch(handler(response.status));
+                    } else {
+                        dispatch(handler(400));
+                    }
```

Comment:
When will the else be called, but not the else-if?

Reply:
The else will be called when we have validation errors.  The controller has to return an OK response to contain the validation errors.  We check to see if there are errors and  we then handle it like a bad request which updates the UI with the field validation errors.  If we had a 401 or 403 or another unexpected error like 500 then the else if case will be used. Otherwise there are field validation errors and we end up in the else case.

Reply:
Ah yes, that makes sense.